### DISCUSSION TITLE
How to organize template overrides?

### DISCUSSION THREAD

**Discussion by NicolasMassart** [squidfunk/mkdocs-material#2129] [Q&A] [answered]
Hi,
I copy the templates from the Python Material package to override and modify on my site.
I take files in the installed Python package directly as it makes sure that I have the correct version. But maybe it's not the best.

1. Should I use files from the package or from the source repos?
2. If I should use the ones from the source repos, should I use the files in `/src` or in `/material` ?
3. Finally, I add the copyright notice from one of the files in `/src` to my overrides. Is that the correct way to do?

See for example this PR https://github.com/hyperledger/besu-docs/pull/573

Thanks.

**Comment by squidfunk** [comment, marked as answer]
1. The files that are packaged are essentially the files in the `material` folder.
2. The files from `src` are commented and may include [some meta tags](https://github.com/squidfunk/mkdocs-material/blob/7e78c4d8b78984bb106372734d1aa3ab2d1fef78/src/base.html#L66) that are replaced during the build process. It's always safer to take the files from `material`.
3. You can change the templates as you like, but note that HTML comments may persist when you don't minify your source and remove them programmatically. This is why the comments of all files are stripped before writing them to the `material` folder.

**Reply by NicolasMassart** [reply]
Ok, and do you require the copyright notice in source code then? It seems normal to me as I use most of your template code and just override some lines.
I will  use minify in the end.

**Reply by squidfunk** [reply]
That's a good question.

From a legal perspective, I'd consider auto-generated content as "binary form", so if you build upon the compiled templates you shouldn't be required to add the copyright, but I'm not a legal expert. Nonetheless, it's MIT, so if there's at some point of the source or distribution files a reference to Material for MkDocs and my name (i.e. the original copyright) it should be fine.

The MIT license states:

> The above copyright notice and this permission notice shall be included in
> all copies or substantial portions of the Software.

I'd not consider overrides substantial, but as said – not a legal expert 😉

**Reply by NicolasMassart** [reply]
Fine I will let the copyright notice in the source code, minify so it's not sent to the browser and anyway there's the material copyright info in the footer. I think we should be fine ;)
Thanks!
